purchases/src/main/kotlin/com/revenuecat/purchases/PostTransactionWithProductDetailsHelper.kt

```diff
@@ -31,18 +32,22 @@ internal class PostTransactionWithProductDetailsHelper(
                     productType = transaction.type,
                     productIds = transaction.productIds.toSet(),
                     onReceive = { storeProducts ->
-                        val purchasedStoreProduct = if (transaction.type == ProductType.SUBS) {
-                            storeProducts.firstOrNull { product ->
-                                product.subscriptionOptions?.let { subscriptionOptions ->
-                                    subscriptionOptions.any { it.id == transaction.subscriptionOptionId }
-                                } ?: false
+                        val purchasedStoreProduct =
+                            // Amazon purchases don't have subscription options
+                            // Amazon is the only store that has marketplace
+                            // We want Google restores to enter in this if, that's why we need to filter Amazon only
+                            if (transaction.type == ProductType.SUBS && transaction.marketplace == null) {
```

Comment:
I thought about just removing the else and checking if `purchasedStoreProduct` is null and in that case doing:

```
storeProducts.firstOrNull { product ->
    product.id == transaction.productIds.firstOrNull()
}
```

But I worry about restored purchases, which also don't have subscription options, and right now we are also getting a `purchasedStoreProduct` for them. That might be another bug, or expected behavior. 

So I went with checking `transaction.marketplace == null` which is a proxy for checking if the store is Google. So if it's Amazon, or consumables, it will go into the else.